src/RA_Core.cpp

```diff
@@ -503,22 +533,15 @@ API void CCONV _RA_ClearMemoryBanks()
 //	}
 //}
 
-API BOOL CCONV _RA_OfferNewRAUpdate(const char* sNewVer)
+static bool RA_OfferNewRAUpdate(const char* sNewVer)
 {
-    char buffer[1024];
-    sprintf_s(buffer, 1024, "Update available!\n"
-        "A new version of %s is available for download at " RA_HOST_URL ".\n\n"
-        "Would you like to update?\n\n"
-        "Current version:%s\n"
-        "New version:%s\n",
-        g_sClientName,
-        g_sClientVersion,
-        sNewVer);
-
-    //	Update last known version:
-    //strcpy_s( g_sKnownRAVersion, 50, sNewVer );
-
-    if (MessageBox(g_RAMainWnd, NativeStr(buffer).c_str(), TEXT("Update available!"), MB_YESNO) == IDYES)
+    std::ostringstream oss;
+    oss << "Would you like to update?\n\n"
+        << "A new version of " << g_sClientName << " is available for download at " << _RA_HostName() << ".\n\n"
+        << "Current version: " << g_sClientVersion << "\n"
+        << "New version: " << sNewVer;
+
+    if (MessageBox(g_RAMainWnd, NativeStr(oss.str()).c_str(), TEXT("Update available!"), MB_YESNO | MB_ICONINFORMATION) == IDYES)
```

Comment:
There was a different comment regarding using `oss.str().c_str()`. I think the way you handled it here is good.